Remove unused`panic-itm` specification from Cargo.toml `dependencies`

charlesincharge opened the pull request:
`panic-itm` was specified both in `dependencies` and in `dev-dependencies` with similar versions. This merges the dependency specifications

burrbull commented:
I don't see any usage of `panic-itm` except examples. So it is `dev-dependencies`.

charlesincharge commented:
@burrbull thanks for pointing that out. I have changed this PR to remove `panic-itm` from `dependencies` then, as it is already specified in `dev-dependencies:
https://github.com/stm32-rs/stm32f407g-disc/blob/431b4ee021371bf0c87e5ccba4627b5819df54f9/Cargo.toml#L40-L44

I have updated the PR name correspondingly.
